src/mappings/events/balances.ts

```diff
@@ -0,0 +1,185 @@
+  value: string;
+}
+
+export interface DecodedEvent {
+  type: string;
+  attributes: Attribute[];
+}
+
+export enum Operation {
+  Increment = 'increment',
+  Decrement = 'decrement',
+}
+
+interface TransactionData {
+  isValidTransaction: boolean;
+  coins: { amount: string; denom: string }[];
+}
+
+const supportedDenominations: Set<string> = new Set(['ubld', 'uist']);
+export const balancesEventKit = () => {
+  function getAttributeValue(data: any, key: string) {
+    const attribute = data.attributes.find(
+      (attr: Attribute) => attr.key === key
+    );
+    return attribute ? attribute.value : null;
+  }
+
+  function getData(cosmosEvent: CosmosEvent) {
+    let dataAlreadyDecoded = true;
+    const value = getAttributeValue(cosmosEvent.event, BALANCE_FIELDS.amount);
+
+    if (!value) {
+      dataAlreadyDecoded = false;
+    }
+
+    const data = dataAlreadyDecoded
+      ? cosmosEvent.event
+      : decodeEvent(cosmosEvent);
+
+    return data;
+  }
+
+  function decodeEvent(cosmosEvent: CosmosEvent): DecodedEvent {
+    const { event } = cosmosEvent;
+
+    const decodedData: DecodedEvent = {
+      type: event.type,
+      attributes: [],
+    };
+
+    event.attributes.forEach((attribute) => {
+      const decodedKey = b64decode(attribute.key);
+      const decodedValue = b64decode(attribute.value);
+
+      decodedData.attributes.push({
+        key: decodedKey,
+        value: decodedValue,
+      });
+    });
+
+    return decodedData;
+  }
+
+  async function addressExists(
+    address: string,
+    denom: string
+  ): Promise<boolean> {
+    const balance = await Balance.getByFields([
+      ['address', '=', address],
+      ['denom', '=', denom],
+    ]);
+
+    if (!balance || balance.length === 0) {
+      return false;
+    }
+    return true;
+  }
+
+  async function createBalancesEntry(
+    address: string,
+    denom: string,
+    primaryKey: string
+  ) {
+    const newBalance = new Balance(primaryKey, address);
+    newBalance.address = address;
+    newBalance.balance = BigInt(0);
+    newBalance.denom = denom;
+    await newBalance.save();
+
+    await createAccountIfNotExists(address);
+
+    logger.info(`Created new entry for address: ${address}`);
+  }
+
+  async function createAccountIfNotExists(address: string): Promise<void> {
+    const account = await Account.get(address);
+    if (!account) {
+      const newAccount = new Account(address);
+      await newAccount.save();
+    }
+  }
+
+  function validateTransaction(amount: string | null): TransactionData {
+    const result: TransactionData = {
+      isValidTransaction: false,
+      coins: [],
+    };
+
+    if (!amount) {
+      return result;
+    }
+    const coins = amount.split(',');
+
+    for (let coin of coins) {
+      for (let denom of supportedDenominations) {
+        if (coin.endsWith(denom)) {
+          result.isValidTransaction = true;
+          result.coins.push({ amount: coin.slice(0, -denom.length), denom });
+        }
+      }
```

Comment:
```suggestion
     result.isValidTransaction = true;
     result.coins.push({ amount: (coin.match(/^\d+/) || [""])[0] , denom: (coin.match(/[a-zA-Z]+.*$/) || [""])[0] });
```

Reply:
`ibc/someHugeNumber` is a fungible token created when a token is recieved from another chain through an IBC transaction.
the number after it might mess with the regex that @toliaqat has suggested